fire/inspectutils.py

```diff
@@ -22,6 +22,27 @@
 
 import IPython
 import six
+from collections import namedtuple
+
+_FullArgTuple = namedtuple('FullArgSpec', ('args', 'varargs', 'varkw',
+  'defaults', 'kwonlyargs', 'kwonlydefaults', 'annotations'))
+
+
+class _FullArgSpec(_FullArgTuple):
```

Comment:
I agree, the tuple is a holdover from python3's inspect.getargspec that I did not reevaluate. 

As this object adds more information, it might be wise to converge it to python 3's inspect.Signature object, which seems to be the [canonical future](https://www.python.org/dev/peps/pep-0362/) of signatures in python. 

Signature has an ordered dict of Parameter (name, kind, default, annotation) objects. I think this might make cleaner parsing in core._ParseFn, and I worry that without this representation your signature and python3's signature will evolve in different directions. It is also probably overpowered for what you need, and an example of small pull requests turning into larger pull requests.

The right answer probably depends on your ideas about the scope of this project. I'm already mucking about here, and happy to modify it as you like.